mozci/push.py

```diff
@@ -33,19 +33,26 @@ class Push:
 
     def __init__(self, revs, branch="autoland"):
         if isinstance(revs, str):
+            self._revs = None
             revs = [revs]
+        else:
+            self._revs = revs
 
-        self.revs = revs
+        rev = revs[0]
         self.branch = branch
-        self._hgmo = HGMO.create(self.rev, branch=self.branch)
-        self.index = BASE_INDEX.format(branch=self.branch, rev=self.rev)
+        self._hgmo = HGMO.create(rev, branch=self.branch)
 
         self._id = None
         self._date = None
+        self.rev = self._hgmo['node']
```

Comment:
> This is what the latest revision of this PR is doing already :)

Isn't this going to trigger a load from hgmo?

Reply:
Yes, I meant it does the `revs` thing on demand already.. The latest push is the one that also only gets `self.rev` from hgmo optionally.